datadog/fwprovider/resource_datadog_downtime_schedule.go

```diff
@@ -0,0 +1,588 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &DowntimeScheduleResource{}
+	_ resource.ResourceWithImportState = &DowntimeScheduleResource{}
+)
+
+type DowntimeScheduleResource struct {
+	Api  *datadogV2.DowntimesApi
+	Auth context.Context
+}
+
+type DowntimeScheduleModel struct {
+	ID                                 types.String                        `tfsdk:"id"`
+	DisplayTimezone                    types.String                        `tfsdk:"display_timezone"`
+	Message                            types.String                        `tfsdk:"message"`
+	MuteFirstRecoveryNotification      types.Bool                          `tfsdk:"mute_first_recovery_notification"`
+	Scope                              types.String                        `tfsdk:"scope"`
+	NotifyEndStates                    types.Set                           `tfsdk:"notify_end_states"`
+	NotifyEndTypes                     types.Set                           `tfsdk:"notify_end_types"`
+	MonitorIdentifier                  *MonitorIdentifierModel             `tfsdk:"monitor_identifier"`
+	DowntimeScheduleRecurrenceSchedule *DowntimeScheduleRecurrenceSchedule `tfsdk:"recurring_schedule"`
+	DowntimeScheduleOneTimeSchedule    *DowntimeScheduleOneTimeSchedule    `tfsdk:"one_time_schedule"`
+}
+
+type MonitorIdentifierModel struct {
+	DowntimeMonitorIdentifierId   types.Int64 `tfsdk:"monitor_id"`
+	DowntimeMonitorIdentifierTags types.Set   `tfsdk:"monitor_tags"`
+}
+
+type DowntimeScheduleRecurrenceSchedule struct {
+	Timezone    types.String        `tfsdk:"timezone"`
+	Recurrences []*RecurrencesModel `tfsdk:"recurrence"`
+}
+type RecurrencesModel struct {
+	Duration types.String `tfsdk:"duration"`
+	Rrule    types.String `tfsdk:"rrule"`
+	Start    types.String `tfsdk:"start"`
+}
+
+type DowntimeScheduleOneTimeSchedule struct {
+	End   types.String `tfsdk:"end"`
+	Start types.String `tfsdk:"start"`
+}
+
+func NewDowntimeScheduleResource() resource.Resource {
+	return &DowntimeScheduleResource{}
+}
+
+func (r *DowntimeScheduleResource) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	providerData, _ := request.ProviderData.(*FrameworkProvider)
+	r.Api = providerData.DatadogApiInstances.GetDowntimesApiV2()
+	r.Auth = providerData.Auth
+}
+
+func (r *DowntimeScheduleResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "downtime_schedule"
+}
+
+func (r *DowntimeScheduleResource) Schema(_ context.Context, _ resource.SchemaRequest, response *resource.SchemaResponse) {
+	response.Schema = schema.Schema{
+		Description: "Provides a Datadog DowntimeSchedule resource. This can be used to create and manage Datadog downtime_schedule.",
```

Comment:
Should we specify this as being in private beta? The output for a non-allowlisted org looks like this

```
        Error: error retrieving DowntimeSchedule
        
          with datadog_downtime_schedule.t1,
          on terraform_plugin_test.tf line 2, in resource "datadog_downtime_schedule" "t1":
           2: resource "datadog_downtime_schedule" "t1" {
        
        403 Forbidden: {"errors":["This feature is currently disabled"]}
```